File: test-projects/simple-multi-file/src/services/userService.ts
import { User, createUser } from '../models/user';

export function getUserService(): string {
	const user: User = createUser('Alice', 'alice@example.com');
	return `User: ${user.name} (${user.email})`;
}

export function getAllUsers(): User[] {
	return [
		createUser('Alice', 'alice@example.com'),
		createUser('Bob', 'bob@example.com')
	];
}

File: go/test-projects/simple-multi-file-go/src/services/userService.go
package services

import (
	"github.com/test/simple-multi-file/models"
)

func GetUserService() string {
	user := CreateUser("Alice", "alice@example.com")
	return /* unsupported expression */
}

func GetAllUsers() []User {
	return []interface{}{CreateUser("Alice", "alice@example.com"), CreateUser("Bob", "bob@example.com")}
}

func main() {
}
